webpack/assets/javascripts/react_app/components/ModelsTable/ModelsTableReducer.js

```diff
@@ -0,0 +1,42 @@
+import Immutable from 'seamless-immutable';
+import {
+  MODELS_TABLE_REQUEST,
+  MODELS_TABLE_SUCCESS,
+  MODELS_TABLE_FAILURE,
+} from './ModelsTableConsts';
+import { STATUS } from '../../constants';
+
+const initState = Immutable({
+  page: 1,
+  perPage: 20,
+  total: 0,
+  sortBy: '',
+  sortOrder: '',
+  results: [],
+  error: null,
+  status: STATUS.PENDING,
+  searchQuery: null,
+});
+export default (state = initState, action) => {
+  switch (action.type) {
+    case MODELS_TABLE_REQUEST:
+      return state.set('status', STATUS.PENDING);
+    case MODELS_TABLE_SUCCESS:
+      return state
+        .set('status', STATUS.RESOLVED)
```

Comment:
TBH  I'm not sure if it really matters for performance, but maybe consider merge(https://github.com/rtfeldman/seamless-immutable) instead of multiple sets?